frontend/src/components/MapView.js

```diff
@@ -19,18 +19,35 @@ const mapStyles = {
 class MapView extends Component {
   constructor(props) {
     super(props);
-    this.state = {
-      allEvents: [],
-      location: undefined,
-      plusCode: props.plusCode,
-      showInfoWindows: true,
-      contents: null,
-      resizeState: false,
-      maxWidth: 100
-    };
+
+    if (this.props.articles[0]) {
```

Comment:
Instead of having to use `this.props.articles[0]` every time, you can create an instance variable `var firstArticle = this.props.articles[0]` and then use that instead. You could apply that in other parts of this PR too